packages/shared/__tests__/define-vars/stylex-define-vars-test.js

```diff
@@ -186,6 +186,156 @@ describe('stylex-define-vars test', () => {
+      },
+      cornerRadius: '10px',
+      fgColor: {
+        default: 'pink',
+      },
+    };
+    const [jsOutput, cssOutput] = styleXDefineVars(defaultVars, {
+      themeName,
+      debug: true,
+    });
+
+    expect(jsOutput).toEqual({
+      __themeName__: classNamePrefix + createHash(themeName),
+      bgColor: `var(--${classNamePrefix + createHash(`${themeName}.bgColor`)})`,
+      bgColorDisabled: `var(--${
+        classNamePrefix + createHash(`${themeName}.bgColorDisabled`)
+      })`,
+      cornerRadius: `var(--${
+        classNamePrefix + createHash(`${themeName}.cornerRadius`)
+      })`,
+      fgColor: `var(--${classNamePrefix + createHash(`${themeName}.fgColor`)})`,
+    });
+
+    expect(cssOutput).toMatchInlineSnapshot(`
+      {
+        "bgColor-x568ih9": {
+          "ltr": ":root, .bgColor-x568ih9{--xgck17p:blue;--xpegid5:grey;--xrqfjmn:10px;--x4y59db:pink;}",
+          "priority": 0,
+          "rtl": null,
+        },
+        "bgColor-x568ih9-1e6ryz3": {
+          "ltr": "@supports (color: oklab(0 0 0)){@media (prefers-color-scheme: dark){:root, .bgColor-x568ih9{--xgck17p:oklab(0.7 -0.3 -0.4);--xpegid5:oklab(0.7 -0.3 -0.4);}}}",
+          "priority": 0.2,
+          "rtl": null,
+        },
+        "bgColor-x568ih9-1lveb7": {
+          "ltr": "@media (prefers-color-scheme: dark){:root, .bgColor-x568ih9{--xgck17p:lightblue;--xpegid5:rgba(0, 0, 0, 0.8);}}",
+          "priority": 0.1,
+          "rtl": null,
+        },
+        "bgColor-x568ih9-bdddrq": {
+          "ltr": "@media print{:root, .bgColor-x568ih9{--xgck17p:white;}}",
+          "priority": 0.1,
+          "rtl": null,
+        },
+        "bgColor-x568ih9-kpd015": {
+          "ltr": "@supports (color: oklab(0 0 0)){:root, .bgColor-x568ih9{--xpegid5:oklab(0.7 -0.3 -0.4);}}",
+          "priority": 0.1,
+          "rtl": null,
+        },
+        "bgColorDisabled-x568ih9": {
+          "ltr": ":root, .bgColorDisabled-x568ih9{--xgck17p:blue;--xpegid5:grey;--xrqfjmn:10px;--x4y59db:pink;}",
+          "priority": 0,
+          "rtl": null,
+        },
+        "bgColorDisabled-x568ih9-1e6ryz3": {
+          "ltr": "@supports (color: oklab(0 0 0)){@media (prefers-color-scheme: dark){:root, .bgColorDisabled-x568ih9{--xgck17p:oklab(0.7 -0.3 -0.4);--xpegid5:oklab(0.7 -0.3 -0.4);}}}",
+          "priority": 0.2,
+          "rtl": null,
+        },
+        "bgColorDisabled-x568ih9-1lveb7": {
+          "ltr": "@media (prefers-color-scheme: dark){:root, .bgColorDisabled-x568ih9{--xgck17p:lightblue;--xpegid5:rgba(0, 0, 0, 0.8);}}",
+          "priority": 0.1,
+          "rtl": null,
+        },
+        "bgColorDisabled-x568ih9-bdddrq": {
+          "ltr": "@media print{:root, .bgColorDisabled-x568ih9{--xgck17p:white;}}",
+          "priority": 0.1,
+          "rtl": null,
+        },
+        "bgColorDisabled-x568ih9-kpd015": {
+          "ltr": "@supports (color: oklab(0 0 0)){:root, .bgColorDisabled-x568ih9{--xpegid5:oklab(0.7 -0.3 -0.4);}}",
```

Comment:
Is this output correct? If looks like the source is a "supports" nested in a "media" for dark mode, but the output is missing the media part.

It's a little difficult to read the test because there are several patterns in the input and a lot of generated content as a result. It might be worth breaking it up so it's easier to check the output by eye

Reply:
Yes, `--xpegid5:` doesn't have the prefix.

Instead the className that defines the variables has the prefix which is likely wrong.